File: patterns/C++/pattern14.cpp
/*
Generate the following pattern
     A
    ABA
   ABCBA
  ABCDCBA
 ABCDEDCBA
*/

#include<bits/stdc++.h>
using namespace std;

void pattern14(int n){
    for(int i=0 ; i<n; i++){
        for(int j=0; j<n-i; j++){
            cout << " ";
        }

        int cnt = 0;
        for(int j=0; j<2*i+1; j++){
            cout << char(65+cnt);

            if(j < (2*i+1)/2) cnt++;
            else cnt --;
        }

        cout << endl;
    }

}


int main(){
    int n;
    cout << "Enter the Number of rows: ";
    cin >> n;

    cout << "\nHere is the generated pattern\n";
    pattern14(n);
}

File: patterns/Python/pattern14.py
"""
Generate the following pattern
     A
    ABA
   ABCBA
  ABCDCBA
 ABCDEDCBA
"""

def pattern14(n):
    for i in range(1, n+1):
        for j in range(n-i):
            print(" ", end="")
        
        cnt = 0
        for j in range(2*i-1):
            print(chr(65+cnt), end="")
            if j < (2*i-1)//2:
                cnt += 1
            else:
                cnt -= 1
                
        print("")
   
#Driver Code 
def main():
    n = int(input("Enter the number of rows: "))
    print("Here is the generated pattern\n")
    pattern14(n)
    
if __name__ == "__main__":
    main()
           